component/main.jsonnet

```
@@ -5,13 +5,20 @@ local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.kyverno;
 
+local nodeSelectionNamespaceLabels = if params.nodeSelectorRole != null then
+  {
+    'openshift.io/node-selector': '',
```

Instead of clearing the node selector completely, you could consider using something like

```suggestion
    'openshift.io/node-selector': 'node-role.kubernetes.io/%s=' % params.nodeSelectorRole,
```